### PR TITLE
Teet 339 select related entities

### PR DESCRIPTION
**Pull request by jkorpelin**
Added functionality to edit related features.

Some parts will still be quite messy since all the showing of lists were heavily bundled with the setup wizard

### CODE REVIEW

File: app/backend/src/clj/teet/project/project_commands.clj [hunk 1]
```diff
@@ -95,46 +97,83 @@ and cadastral units"
...
+            (for [id-to-remove to-be-removed]
+              [:db/retract project-eid
+               link-attribute-kw id-to-remove])
+            (for [id-to-add to-be-added]
+              [:db/add project-eid
+               link-attribute-kw id-to-add])))))
+
+(defcommand :thk.project/update-restrictions
+  {:doc "Update project related restrictions"
+   :context {:keys [user db conn]}
```
Comment (tatut) [inline app/backend/src/clj/teet/project/project_commands.clj]: dont need to destructure `conn` from context anymore, same in the below command

File: app/frontend/src/cljs/teet/project/project_controller.cljs [hunk 2]
```diff
@@ -220,17 +220,44 @@
       (->NavigateToStep step-label))))
 
 (extend-protocol t/Event
-  FetchRelatedFeaturesResponse
-  (process-event [{:keys [result-path geojson-path response]} app]
-    (let [geojson (js/JSON.parse response)
+  FetchFeatureCandidatesResponse
+  (process-event [{:keys [candidate-type response]} app]
+    (let [result-path (if (= candidate-type :restrictions)
```
Comment (tatut) [inline app/frontend/src/cljs/teet/project/project_controller.cljs]: these could be put to a "config map" based on candidate type and destructured from there

`(def candidate-paths {:restrictions {:result-path ...} :cadastral-units {:result-path ...}})` 

and uses `(let [{:keys [result-path ...]} (candidate-paths candidate-type)`

File: app/frontend/src/cljs/teet/project/project_layers.cljs [hunk 3]
```diff
@@ -173,20 +187,26 @@
...
 
-(defn selected-restrictions [{{checked-restrictions-geojson :checked-restrictions-geojson
+(defn selected-restrictions [{{:keys [query]} :app
+                              {checked-restrictions-geojson :checked-restrictions-geojson
                                setup-step :setup-step} :project}]
   (when (and checked-restrictions-geojson
-             (= setup-step "restrictions"))
+             (or
+               (= (:configure query) "restrictions")
+               (= setup-step "restrictions")))
```
Comment (outamaa) [inline app/frontend/src/cljs/teet/project/project_layers.cljs]: This pattern is repeated quite a few times. I guess this has to do with getting things working in both the wizard and the new views?

Comment (jkorpelin) [reply app/frontend/src/cljs/teet/project/project_layers.cljs]: Yes, would be great if this could somehow be made more clear.

**Pull request merged**